braindecode/models/conformer.py

```diff
@@ -0,0 +1,279 @@
+        self.num_heads = num_heads
+        self.keys = nn.Linear(emb_size, emb_size)
+        self.queries = nn.Linear(emb_size, emb_size)
+        self.values = nn.Linear(emb_size, emb_size)
+        self.att_drop = nn.Dropout(dropout)
+        self.projection = nn.Linear(emb_size, emb_size)
+
+    def forward(self, x: Tensor, mask: Tensor = None) -> Tensor:
+        queries = rearrange(
+            self.queries(x), "b n (h d) -> b h n d", h=self.num_heads
+        )
+        keys = rearrange(
+            self.keys(x), "b n (h d) -> b h n d", h=self.num_heads
+        )
+        values = rearrange(
+            self.values(x), "b n (h d) -> b h n d", h=self.num_heads
+        )
+        energy = torch.einsum("bhqd, bhkd -> bhqk", queries, keys)
+        if mask is not None:
+            fill_value = torch.finfo(torch.float32).min
+            energy.mask_fill(~mask, fill_value)
+
+        scaling = self.emb_size ** (1 / 2)
+        att = F.softmax(energy / scaling, dim=-1)
+        att = self.att_drop(att)
+        out = torch.einsum("bhal, bhlv -> bhav ", att, values)
+        out = rearrange(out, "b h n d -> b n (h d)")
+        out = self.projection(out)
+        return out
+
+
+class ResidualAdd(nn.Module):
+    def __init__(self, fn):
+        super().__init__()
+        self.fn = fn
+
+    def forward(self, x, **kwargs):
+        res = x
+        x = self.fn(x, **kwargs)
+        x += res
+        return x
+
+
+class FeedForwardBlock(nn.Sequential):
+    def __init__(self, emb_size, expansion, drop_p):
+        super().__init__(
+            nn.Linear(emb_size, expansion * emb_size),
+            nn.GELU(),
+            nn.Dropout(drop_p),
+            nn.Linear(expansion * emb_size, emb_size),
+        )
+
+
+class GELU(nn.Module):
+    def forward(self, input: Tensor) -> Tensor:
+        return input * 0.5 * (1.0 + torch.erf(input / math.sqrt(2.0)))
+
+
+class TransformerEncoderBlock(nn.Sequential):
+    def __init__(self, emb_size, att_heads, att_drop, forward_expansion=4):
+        super().__init__(
+            ResidualAdd(
+                nn.Sequential(
+                    nn.LayerNorm(emb_size),
+                    MultiHeadAttention(emb_size, att_heads, att_drop),
+                    nn.Dropout(att_drop),
+                )
+            ),
+            ResidualAdd(
+                nn.Sequential(
+                    nn.LayerNorm(emb_size),
+                    FeedForwardBlock(
+                        emb_size, expansion=forward_expansion, drop_p=att_drop
+                    ),
+                    nn.Dropout(att_drop),
+                )
+            ),
+        )
+
+
+class TransformerEncoder(nn.Sequential):
+    def __init__(self, att_depth, emb_size, att_heads, att_drop):
+        super().__init__(
+            *[
+                TransformerEncoderBlock(emb_size, att_heads, att_drop)
+                for _ in range(att_depth)
+            ]
+        )
+
+
+class ClassificationHead(nn.Sequential):
```

Comment:
Use `nn.Module ` instead if you are writing your model in that way.
```suggestion
class ClassificationHead(nn.Module):
```

Reply:
Okay, thank you for the modification.